backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/picking/job_schedule/service/commands/PickingJobScheduleAutoAssignCommand.java

```diff
@@ -0,0 +1,343 @@
+				.collect(ImmutableList.toImmutableList());
+
+		if (workplacesWithCapacity.isEmpty())
+		{
+			return;
+		}
+
+		final ImmutableSet<WarehouseId> warehouseIds = workplacesWithCapacity.stream()
+				.map(Workplace::getWarehouseId)
+				.collect(ImmutableSet.toImmutableSet());
+
+		final ImmutableList<ShipmentSchedule> allEligibleShipmentSchedules = shipmentScheduleRepository.getBy(
+				ShipmentScheduleQuery.builder()
+						.warehouseIds(warehouseIds)
+						.preparationDate(request.getPreparationDate())
+						.fromCompleteOrderOrNullOrder(true)
+						.includeWithQtyToDeliverZero(false)
+						.includeProcessed(false)
+						.build())
+				.stream()
+				.filter(sched -> sysConfigBL.getBooleanValue(SYSCONFIG_CARRIER_PRODUCT_REQUIRED, false) || sched.getCarrierProductId() != null)
+				.collect(ImmutableList.toImmutableList());
+
+		if (allEligibleShipmentSchedules.isEmpty())
+		{
+			return;
+		}
+
+		final ImmutableSet<ShipmentScheduleId> allEligibleShipmentScheduleIds = allEligibleShipmentSchedules.stream()
+				.map(ShipmentSchedule::getId)
+				.collect(ImmutableSet.toImmutableSet());
+
+		final ImmutableSet<WorkplaceId> workplaceIds = workplacesWithCapacity.stream()
+				.map(Workplace::getId)
+				.collect(ImmutableSet.toImmutableSet());
+
+		// TODO think about / clarify what todo if qtyScheduled != qtyToDeliver (currently additional qty would be scheduled after old schedule is processed)
+		final List<PickingJobSchedule> existingSchedules = pickingJobScheduleRepository.stream(
+				PickingJobScheduleQuery.builder()
+						.onlyShipmentScheduleIds(allEligibleShipmentScheduleIds)
+						.workplaceIds(workplaceIds)
+						.isProcessed(false) //TODO check if we could run into processed, but qtyToDeliver not yet updated
+						.build()
+		).collect(ImmutableList.toImmutableList());
+
+		final ImmutableSet<ShipmentScheduleId> alreadyAssignedIds = existingSchedules.stream()
+				.map(PickingJobSchedule::getShipmentScheduleId)
+				.collect(ImmutableSet.toImmutableSet());
+
+		final ImmutableList<ShipmentSchedule> allUnscheduledShipmentScheduleIds = allEligibleShipmentSchedules.stream()
+				.filter(sched -> !alreadyAssignedIds.contains(sched.getId()))
+				.collect(ImmutableList.toImmutableList());
+
+		final Map<WorkplaceId, Integer> assignedCountPerWorkplace = new HashMap<>();
+		existingSchedules.forEach(schedule ->
+				assignedCountPerWorkplace.merge(schedule.getWorkplaceId(), 1, Integer::sum)
+		);
+
+		final ImmutableSet<ProductId> productIds = allUnscheduledShipmentScheduleIds.stream()
+				.map(ShipmentSchedule::getProductId)
+				.collect(ImmutableSet.toImmutableSet());
+		final ImmutableMap<ProductId, Product> productsById = productRepository.getMapByIds(productIds);
+
+		final ImmutableMap<OrderId, BigDecimal> totalQtyToDeliverByOrderId = allUnscheduledShipmentScheduleIds.stream()
+				.collect(ImmutableMap.toImmutableMap(
+						ShipmentSchedule::getOrderId,
+						schedule -> schedule.getQuantityToDeliver().toBigDecimal(),
```

Comment:
why not Qty to deliver (effective) minus de.metas.inoutcandidate.model.I_M_ShipmentSchedule#getQtyScheduledForPicking ?

Reply:
> also, why groupping by OrderId?

Logic from Sql provided as RE
```sql
shipment_type AS (
    SELECT
        us.*,
        CASE
            --WHEN COUNT(*) OVER (PARTITION BY us.c_orderline_id) = 1 THEN 'SG'
            --ELSE 'MP'
            WHEN COALESCE(SUM(us.qtytodeliver) OVER (PARTITION BY us.c_order_id), 0) = 1 THEN  'SG'
        ELSE 'MP'
        END AS picking_type
    FROM unscheduled_shipments us
),
```